node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/maintenance/acl/AclMaintainer.java

```diff
@@ -63,15 +64,13 @@ private void applyAcl(ContainerName containerName, Acl acl) {
         }
         final Command flush = new FlushCommand(Chain.INPUT);
         final Command rollback = new PolicyCommand(Chain.INPUT, Action.ACCEPT);
-        log.info("Start modifying ACL rules for " + containerName.asString());
         try {
-            log.debug("Running ACL command '" + flush.asString() + "'");
-            dockerOperations.executeCommandInNetworkNamespace(containerName, flush.asArray(IPTABLES_COMMAND));
-            acl.toCommands().forEach(command -> {
-                log.debug("Running ACL command '" + command.asString() + "' for " + containerName.asString());
-                dockerOperations.executeCommandInNetworkNamespace(containerName,
-                                                                  command.asArray(IPTABLES_COMMAND));
-            });
+            String commands = Stream.concat(Stream.of(flush), acl.toCommands().stream())
+                    .map(command -> command.asString(IPTABLES_COMMAND))
+                    .collect(Collectors.joining("; "));
+
+            log.debug("Running ACL command '" + commands + "' in " + containerName.asString());
+            dockerOperations.executeCommandInNetworkNamespace(containerName, "/bin/sh", "-c", commands);
```

Comment:
Any practical reason for doing this change? 

This may not work for large rulesets where the joined commands exceed the kernels maximum allowed command length (`ARG_MAX`).

Reply:
You could write the commands to a file and then execute that as a script.

Reply:
In short, the goal is to reduce load on `node-admin`, `docker`/`kernel`. Everytime there is a burst of commands, there is an increased chance of something hanging indefinitely, and an ACL setup for a single node executes 8+ commands, this will reduce it to a single execution.

As for the size, I don't think that'll be a problem; 
* 246 bytes initial 
* 38 + length of the IPv6 address, per IP
* 28 bytes for final `REJECT`
Even if we send the entire IPv6 without "::" shorthand (39 bytes per address), would need less than 8KB for the entire command with 100 addresses in ACL, this is way below default(?) `ARG_MAX` at 2097152 bytes

Reply:
Why is there an increased chance of something hanging? This approach will execute just as many `iptables` commands, just inside another shell. To reduce the number of commands you would have to use `ipset` instead.

The only difference I see here is that `nsenter` will be executed once since there's only one call to `executeCommandInNetworkNamespace`, but the burst will be the same.

Reply:
Yes,should have been more clear, the goal is to reduce the number of calls where enter container namespace, such as `docker exec` and `nsenter`. They become unreliable under high load and can randomly fail. Additionally the container can go down at any time, and calling once vs multiple times reduces number of errors due to container going down in the middle of the chain of commands.

Reply:
I see, let's try it then.